Jenkinsfile

```diff
@@ -33,17 +25,21 @@ pipeline {
 			steps{
 				script{
 					withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
-						def slice_test_snapshot_dump = "${env.SLICE_TEST}_${currentRelease}_snapshot.dump"
-						def slice_previous_snapshot_dump = "${env.SLICE_TEST}_${previousRelease}_snapshot.dump.gz"
-						// Retrieve snapshot DB from S3.
-						sh "aws s3 --no-progress cp ${env.S3_RELEASE_DIRECTORY_URL}/${previousRelease}/update_stable_ids/databases/$slice_previous_snapshot_dump ."
-						sh "mysql -u$user -p$pass -e \'drop database if exists ${env.SLICE_PREVIOUS}; create database ${env.SLICE_PREVIOUS}\'"
-						sh "zcat  $slice_previous_snapshot_dump | mysql -u$user -p$pass ${env.SLICE_PREVIOUS}"
-						sh "rm $slice_previous_snapshot_dump"
-						sh "mysqldump -u$user -p$pass ${env.SLICE_TEST} > $slice_test_snapshot_dump"
-						sh "gzip -f $slice_test_snapshot_dump"
-						sh "mysql -u$user -p$pass -e \'drop database if exists ${env.SLICE_CURRENT}; create database ${env.SLICE_CURRENT}\'"
-						sh "zcat  ${env.SLICE_TEST}_${currentRelease}_snapshot.dump.gz | mysql -u$user -p$pass ${env.SLICE_CURRENT}"
+						
+						// Replace 'slice_previous' db with snapshot slice DB from S3.
+						def previousReleaseVersion = utils.getPreviousReleaseVersion()
+						def slicePreviousSnapshotDump = "${env.SLICE_TEST_DB}_${previousReleaseVersion}_snapshot.dump.gz"
+						// Retrieve previous releases' snapshot slice DB from S3.
+						sh "aws s3 --no-progress cp ${env.S3_RELEASE_DIRECTORY_URL}/${previousReleaseVersion}/update_stable_ids/databases/${slicePreviousSnapshotDump} ."
+						utils.replaceDatabase("${env.SLICE_PREVIOUS_DB}", "${slicePreviousSnapshotDump}")
```

Comment:
Does this do a back-up of the database being overwritten so it isn't lost just in case of a mistake/issue and it needing to be restored?

Reply:
It doesn't, because at the times this would be used there would already be an existing copy of that database. I don't have any problem incorporating a database dump aspect to this, but just decided on this to reduce redundant backups

Reply:
>  there would already be an existing copy of that database

Where in the process does that existing copy get created? Where does it exist at _this_ point in the process?

Reply:
At _this _point_, it is just the final release database from the previous release. That database exists, probably in duplicate or triplicate, in S3

Reply:
It's the "probably" that concerns me...

Reply:
I will have this step take a database dump